### PR TITLE
Resolve incompatibility of TwoModeQuadratureCorrelation class with Qu…

### LINKED ISSUE #2546: Bug in distributions.py concerning Qobj.data and Qobj.data_as()
### Bug Description

Hi, I think there is an bug in the distributions.py file. The functions `update_psi()` and  `update_rho()` of the class `TwoModeQuadratureCorrelation` both have a `Qobj.data[i,j]` type statement as was the norm before qutip5. I encountered this issue when code from my PC would not run on my MacBook. I could fix it by just replacing the `psi.data[i,0]` with a `psi.data_as()[i,0]` (and analogous with `rho.data`) in the distributions.py file as noted in the change log. 

### Code to Reproduce the Bug

```shell
import qutip as qt
a = qt.tensor(qt.destroy(5), qt.qeye(5))
b = qt.tensor(qt.qeye(5), qt.destroy(5))
z = 0.9
rho=qt.squeezing(a,b,z)*qt.tensor(qt.fock(5,0),qt.fock(5,0))
p=qt.TwoModeQuadratureCorrelation(rho,theta1=0,theta2=0)
p.visualize()
```

### Code Output

```shell
TypeError                                 Traceback (most recent call last)
Cell In[69], line 44
     39 print("P quadrature correlation:", p_corr)
     41 # Ensure the correct dimensions are used by TwoModeQuadratureCorrelation
     42 # Calculating two-mode quadrature correlation (if this is the function you mean)
     43 # This is a placeholder for any specific function you're calling:
---> 44 corr = qt.TwoModeQuadratureCorrelation(two_mode_squeezed_state)
     45 # print("Two-mode quadrature correlation:", corr)
File ~/Library/CloudStorage/OneDrive-ETHZurich/PythonScripts/Qutip/TMS_Kerr_simulations/.venv/lib/python3.12/site-packages/qutip/distributions.py:283, in TwoModeQuadratureCorrelation.__init__(self, state, theta1, theta2, extent, steps)
    280 self.theta1 = theta1
    281 self.theta2 = theta2
--> 283 self.update(state)
File ~/Library/CloudStorage/OneDrive-ETHZurich/PythonScripts/Qutip/TMS_Kerr_simulations/.venv/lib/python3.12/site-packages/qutip/distributions.py:291, in TwoModeQuadratureCorrelation.update(self, state)
    286 """
    287 calculate probability distribution for quadrature measurement
    288 outcomes given a two-mode wavefunction or density matrix
    289 """
    290 if isket(state):
--> 291     self.update_psi(state)
    292 else:
    293     self.update_rho(state)
...
    315         i = state_number_index([N, N], [n1, n2])
--> 316         p += kn1 * kn2 * psi.data[i, 0]
    318 self.data = abs(p) ** 2
TypeError: 'qutip.core.data.dense.Dense' object is not subscriptable
```

### Expected Behaviour

(<Figure size 800x600 with 2 Axes>,
 <Axes: xlabel='$X_1(\\theta_1)$', ylabel='$X_2(\\theta_2)$'>)
image

### Your Environment

```shell
QuTiP Version:      5.0.4
Numpy Version:      2.1.2
Scipy Version:      1.14.1
Cython Version:     None
Matplotlib Version: 3.9.2
Python Version:     3.12.5
Number of CPUs:     11
BLAS Info:          Generic
OPENMP Installed:   False

vs. on PC:

QuTiP Version:      4.7.2
Numpy Version:      1.25.2
Scipy Version:      1.11.1
Cython Version:     None
Matplotlib Version: 3.7.1
Python Version:     3.11.4
Number of CPUs:     24
BLAS Info:          INTEL MKL
OPENMP Installed:   False
```

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by quantum-menace**
…TiP 5.0.0 and above

Adjust update_psi and update_rho functions of TwoModeQuadratureCorrelation class to be compatible with QuTiP 5.0.0. Change Qobj.data to Qobj.data_as() for both functions.

**Checklist**
Thank you for contributing to QuTiP! Please make sure you have finished the following tasks before opening the PR.

- [x] Include the changelog in a file named: `doc/changes/<PR number>.<type>` 'type' can be one of the following: feature, bugfix, doc, removal, misc, or deprecation (see [here](http://qutip.org/docs/latest/development/contributing.html#changelog-generation) for more information).

Delete this checklist after you have completed all the tasks. If you have not finished them all, you can also open a [Draft Pull Request](https://github.blog/2019-02-14-introducing-draft-pull-requests/) to let the others know this on-going work and keep this checklist in the PR description.

**Description**
Adjust update_psi and update_rho functions of TwoModeQuadratureCorrelation class to be compatible with QuTiP 5.0.0. Change Qobj.data to Qobj.data_as() for both functions.

**Related issues or PRs**
fix #2546

### CODE REVIEW

**Review by Ericgig — COMMENTED** [review]
Thank you for the contribution.

There are 2 other uses of `.data[...]` in other distribution classes, could you also fix them? (Line 384 and 422.)

There are also no test for it. Could you add the failing code you have in #2546 in a test function.  You could put it in `test_animation` or `test_visualization`. No need to test that the output is right, just that is run without raising issues.

**Comment by Ericgig** [comment]
Thank you for the tests. 
These tests will already ensure future changes don't break anything so we are quite grateful for them as is.

We treat warnings as error during automated tests, so the casting warnings need to be fixed.
You change `self.data` to `self.data.real` in all 3 `visualize_...` functions. It will show same plot without the warnings.

Maybe if there are interest, we could add an option as to what to show (real, imag, abs, ...), but not in this PR.

**Comment by quantum-menace** [comment]
I adjusted 3 `visualize_...` functions as suggested. No more warnings. Thanks for the input!

File: qutip/distributions.py [hunk 1]
```diff
@@ -125,11 +125,11 @@ def visualize_2d_colormap(self, fig=None, ax=None, figsize=(8, 6),
             fig, ax = plt.subplots(1, 1, figsize=figsize)
 
         if cmap is None:
-            cmap = mpl.cm.get_cmap('RdBu')
+            cmap = mpl.colormaps.get_cmap('RdBu')
```
Comment (Ericgig) [inline qutip/distributions.py:128]: This is not supported by old matplotlib versions.
```suggestion
            cmap = mpl.colormaps['RdBu']
```
works from at least matplotlib 3.5 which is the oldest we test.

File: qutip/distributions.py [hunk 2]
```diff
@@ -152,12 +152,12 @@ def visualize_2d_surface(self, fig=None, ax=None, figsize=(8, 6),
             ax = Axes3D(fig, azim=-62, elev=25)
 
         if cmap is None:
-            cmap = mpl.cm.get_cmap('RdBu')
+            cmap = mpl.colormaps.get_cmap('RdBu')
```
Comment (Ericgig) [inline qutip/distributions.py:155]:
```suggestion
            cmap = mpl.colormaps['RdBu']
```

**Review by Ericgig — APPROVED** [review]
Thank you

**Pull request merged**